src/main/java/kr/codesqaud/cafe/controller/UserController.java

```diff
@@ -9,22 +11,26 @@
 
 @Controller
 public class UserController {
-    private UserService userService = new UserService();
+    private UserService userService;
```

Comment:
```suggestion
    private final UserService userService;
```

\+ `final` 키워드를 사용함으로써 얻을 수 있는 장점에 대해서도 생각해보세요 :)

Reply:
넵! 
(final 키워드의 장점에 대해서는 전에 찾아본 적이 있었습니다:
결론이 '성능 향상까지 되기 때문에, 안 되는 경우를 빼고는 전부 다 final를 깔아주면 좋다!' 라는 정도까지 였었는데요(!!))

제가 final에 대해 알고 있는 것은 원래는 '상수를 만든다' 였는데요, 정확한 것은 아니었습니다. (int 변수의 경우라면 맞겠습니다만) 지금은 '참조 변수를 교체하지 못하게 한다'인 것 같고, 그래서 배열의 경우 그 안의 요소를 바꿀 수 있다고 알고 있습니다.

(혹시 잘못 알고 있는지 모르겠습니다만) 심지어 성능 향상까지 되기 때문에 안 되는 경우 빼고는 다 final을 거는 것이 좋을 정도면 당연히 기본적으로 해야 할 것 같구요, 이 코드의 경우에서 필드 변수(UserService)가 교체가 되야 할 필요가 없는 것이 확실하다면 당연히 final을 시켜 주는 것이 좋을 것 같습니다.

(일단 제 머릿속에서 final의 사용해야 할 순간은 '안 되는 경우 빼고 무조건 다'라고 기억하고 있습니다 😅) 맞게 알고 있는 것인지 쪼금 불안합니다 ㅎㅎㅎㅎㅎ

Reply:
final 키워드의 성능 향상까지 인지하고 계셨다니! 학습하신 내용들이 하나하나 잘 정리되면 더 좋겠네요 :)

힌트를 드리자면 지금 단계에서는 `초기화`, `불변성`, `컴파일 에러` 키워드를 중심으로 final 키워드에 대해 학습해보세요!

Reply:
🤩대-박🤩 엄청 매우 감사합니다....!!!!

Reply:
블로그를 참고해서 조금 더 정리해 보았습니다 - final 키워드) 😄 
final 변수: 초기화 이후 '재할당'이 불가능하게 합니다 -> 만약에 시도하면: '컴파일 에러 발생'
      ==> 단순 변수(상수)를 만드는 경우 초기화는 필드변수로 하면 되는데/ 그 외엔 생성자에서 한다고 합니다(상수 외의 객체 변수 같은 경우도 그냥 필드 변수로 하면 왜 안되는지가 궁금해졌습니다)
      일반적으로 static final을 하는 이유에 대해서는 지난 번엔 이해했었는데 지금 다시 보니 약간 이해가 안 되는 듯해 나중에 다시 봐야 할 듯 합니다😅 

final 메소드: '재정의'를 불가능하게 합니다. -> 만약에 시도하면: '컴파일 에러 발생'

final과 불변성: 원시타입은 final을 붙여서 불변이 될 수 있지만 참조타입은 final로 불변이 되지 않습니다. 
(=> 더 복잡한 방법이 필요합니다😇)